manifests/net_process_log.pp

```diff
@@ -0,0 +1,60 @@
+# @summary Logs user processes and open network connections to syslog
+#
+# Logs user processes and open network connections to syslog
+#
+# @param ignore_users
+#   String to pass to net_process_log.pl script as -i argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#
+# @param ps_arg
+#   String to pass to net_process_log.pl script as --psarg argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#
+# @param ss_arg
+#   String to pass to net_process_log.pl script as --ssarg argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#   
+# @param ss_filter
+#   String to pass to net_process_log.pl script as --ssfilter argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#
+# @param minute_interval
+#   Value to use in the minute field of the cron task 
+#
+# @example
+#   include profile_audit::net_process_log
+class profile_audit::net_process_log (
+  String $ignore_users,
+  String $ps_arg,
+  String $ss_arg,
+  String $ss_filter,
+  String $minute_interval,
+) {
+
+  file { '/root/scripts/net_process_log.pl':
+    ensure => 'present',
+    mode   => '0750',
+    owner  => 'root',
+    group  => 'root',
+    source => "puppet:///modules/${module_name}/net_process_log.pl",
+  }
+
+  # Setup any non-default options, otherwise *_option stays blank
+  # so we use default(s) from script
+  if $ignore_users != "" { $ignore_users_option = "-i \"${ignore_users}\"" }
+  if $ps_arg != "" { $ps_arg_option = "--psarg \"${ps_arg}\"" }
+  if $ss_arg != "" { $ss_arg_option = "--ssarg \"${ss_arg}\"" }
+  if $ss_filter != "" { $ss_filter_option = "--ssfilter \"${ss_filter}\"" }
+
+  cron { 'net_process_log':
+    user        => 'root',
+    minute      => $minute_interval,
+    hour        => '*',
+    month       => '*',
+    weekday     => '*',
+    monthday    => '*',
+    environment => ['SHELL=/bin/sh', ],
```

Comment:
Would there any rationale to let all of the CRON parameters to be set from a Hash parameter, in particular the time scheduling ones? i.e. Can it steal your CRON logic from https://github.com/ncsa/puppet-profile_metrics_alerting/blob/bsper2/SVC-5148/alert-pausing/manifests/alert_cycle.pp ?

Or does this really only make sense to run every few minutes?

Reply:
It doesn't make sense to run it less than once an hour so that's why I was hard-coding most of the params and only making the minute tunable.